flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/internals/ShardConsumer.java

```diff
@@ -235,6 +235,9 @@ public void run() {
 						subscribedShard.getShard().getHashKeyRange().getStartingHashKey(),
 						subscribedShard.getShard().getHashKeyRange().getEndingHashKey());
 
+					recordBatchSizeBytes = 0L;
```

Comment:
Actually, why don't you just initialize these variable here rather than above.  You can also move the block that does the adaptive computation below here rather than above.

Reply:
Yes makes sense. Changed.